can't install width Centos6.5&canvas@2.0.0-alpha.4

I flow wiki do everything i can,but it's still throw err ：

make: *** [Release/obj.target/canvas/src/backend/Backend.o] 错误 1
make: Leaving directory `/wacao-app/wacao-service/node_modules/canvas/build'
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:276:23)
gyp ERR! stack     at emitTwo (events.js:106:13)
gyp ERR! stack     at ChildProcess.emit (events.js:194:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:215:12)
gyp ERR! System Linux 2.6.32-573.22.1.el6.x86_64
gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /wacao-app/wacao-service/node_modules/canvas
gyp ERR! node -v v7.7.1
gyp ERR! node-gyp -v v3.5.0
gyp ERR! not ok
npm ERR! Linux 2.6.32-573.22.1.el6.x86_64
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "install" "--save-dev" "canvas@2.0.0-alpha.4"
npm ERR! node v7.7.1
npm ERR! npm  v4.1.2
npm ERR! code ELIFECYCLE

npm ERR! canvas@2.0.0-alpha.4 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the canvas@2.0.0-alpha.4 install script 'node-gyp rebuild'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the canvas package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs canvas
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls canvas
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /wacao-app/wacao-service/npm-debug.log

## Your Environment
* Version of node-canvas (e.g. ^2.0.0-alpha.6):
* Environment (e.g. node 7.7.1 on Centos6.5):

--- a/src/backend/Backend.h
+++ b/src/backend/Backend.h
@@ -50,7 +50,14 @@ class Backend : public Nan::ObjectWrap
     virtual void setHeight(int height);
 
     // Overridden by ImageBackend. SVG and PDF thus always return INVALID.
-    virtual cairo_format_t getFormat() { return CAIRO_FORMAT_INVALID; }
+    virtual cairo_format_t getFormat() {
+#ifndef CAIRO_FORMAT_INVALID
+      // For old Cairo (CentOS) support
+      return static_cast<cairo_format_t>(-1);
+#else
+      return CAIRO_FORMAT_INVALID;
+#endif
+    }
 };
 
 

